deviseを使って、ユーザー認証を実装しました。

naomichi-h opened the pull request:
deviseを使って、ユーザー認証を実装しました。
また、ユーザー一覧＆詳細ページも作成しました。
rubocopは通過済みです。
確認のほど、よろしくお願いいたします。

naomichi-h committed:
deviseを使用し、ユーザー認証の雛形を作成しました

naomichi-h committed:
update:アカウント編集画面で編集できる項目を追加しました

名前、郵便番号、住所、プロフィール文を編集できるように変更。
アカウント編集後、アカウント編集画面にリダイレクトするように変更。
アカウント編集画面からアプリのホーム画面に遷移できるようにリンクを作成。

naomichi-h committed:
ユーザー一覧画面とユーザー詳細画面を追加しました

naomichi-h committed:
update:日本語対応しました

devise-i18nを導入し、viewの再生成と再編集を行いました。
rubocop-fjord通過ずみです。

yoshitsugu commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
このファイルは不要そうなので、削除してしまって大丈夫です！ :ok_man: 
他にも不要そうなファイルは削除おねがいします！

yoshitsugu commented on app/controllers/users/confirmations_controller.rb:
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
```
confirmable使っていないので、この辺も不要かなーと思います。

yoshitsugu commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Users::RegistrationsController < Devise::RegistrationsController
+  # before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+  # def new
+  #   super
+  # end
+
```
不要なコメントも消しちゃってよいですよ〜

yoshitsugu commented on app/views/devise/mailer/confirmation_instructions.html.erb:
```diff
@@ -0,0 +1,5 @@
+<p><%= t('.greeting', recipient: @email) %></p>
```
confirmation不要なので、この辺も不要です〜

yoshitsugu commented on config/routes.rb:
```diff
@@ -1,4 +1,14 @@
+    registrations: "users/registrations"
+  }
+
+  resources :users, :only => [:index, :show]
```
すごく細かいですし個人的な好みなので、無視してしまってかまわないですが、

```suggestion
  resources :users, only: [:index, :show]
```

の方がスッキリするかなーと思います

yoshitsugu reviewed (commented):
コードとしてはOKそうです！
不要ファイルがたくさんありそうなので、最小限どれが必要そうか、確認してみてください〜

naomichi-h committed:
fix:レビューを受けて修正を行いました

不要なscssファイルを削除しました
confirmableに関するファイルを削除しました
omniauthに関するファイルを削除しました
不要なコメントを削除しました

naomichi-h commented:
レビューありがとうございます❗️
指摘を受けた箇所の修正を行いました。

「不要なファイルを削除しよう」
「不要なコメントを消そう」
に関しては、わかる範囲で対応しました...。

確認のほど、よろしくお願いします❗️

yoshitsugu commented on app/helpers/users_helper.rb:
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+module UsersHelper
+end
```
これも不要です〜

naomichi-h replied:
三つあるhelperファイルのうち、`application_helper.rb`と`books_helper.rb`は必要で、`users_helper.rb`は不要だというのがわからないのですが、どういう理由なのでしょうか？

yoshitsugu replied:
@naomichi-h 差分としてでているのがusers_helperだったので、ここにコメントしましたが、 `books_helper` も使ってなければ削除してしまってよいですよ〜。 application_helepr.rbは、大本のhelperになりますので、あったほうがよいと思います！

naomichi-h replied:
なるほど、差分からの判断だったのですね😊理解できました！
users_helper.rbのみ消そうと思います！

yoshitsugu commented on test/models/user_test.rb:
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
このファイルもテストしないなら不要かと思います〜

naomichi-h replied:
テストに関しては、この後のプラクティスでやるとのことで、現在、railsのテストに対する知識がほぼ無い状態です。
そのため、テストに関する部分の要・不要等が全くわからず、このファイルが不要な理由もわかりません...。
今の時点で先にテストのプラクティスをやるべきでしょうか？

yoshitsugu replied:
@naomichi-h いえ、それであればとりあえず無視でもいいですよ :+1:

naomichi-h replied:
わかりました！では、指摘いただいたuser_test.rbの削除のみ行おうと思います😊

yoshitsugu commented on db/migrate/20210611070344_add_detail_to_users.rb:
```diff
@@ -0,0 +1,10 @@
+
+class AddDetailToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :name, :string
```
実務で使う場合は、つけられるものは `null: false` つけておくと事故が少なくてよいです :+1:

yoshitsugu reviewed (commented):
いくつかコメントしました〜

naomichi-h committed:
fix: レビューを受けて修正を行いました(2)
不要なhelperファイルを削除しました
不要なテストファイルを削除しました
migrationファイルにnull:falseを追加しました

naomichi-h commented:
再度の確認ありがとうございます🙇‍♂️
指摘いただいた箇所の修正を行いましたので、確認よろしくお願いいたします。

yoshitsugu commented on db/migrate/20210611070344_add_detail_to_users.rb:
```diff
@@ -2,9 +2,9 @@
-    add_column :users, :zip, :integer
-    add_column :users, :address, :string
-    add_column :users, :profile, :text
+    add_column :users, :name, :string, null: false, default: ''
+    add_column :users, :zip, :integer, null: false, default: ''
+    add_column :users, :address, :string, null: false, default: ''
+    add_column :users, :profile, :text, null: false, default: ''
```
`default: ''` は何も指定しなかったら空文字列がはいってしまうので、これも厳密に値がはいってほしいものにはつけないほうがいいですね。とはいえ今回はOKです！

yoshitsugu reviewed (approved):
OKです！1点細かいところだけコメントしました〜
